src/ansible_navigator/configuration_subsystem/defs_presentable.py

```diff
@@ -0,0 +1,180 @@
+        """
+        return str(self.current_value)
+
+    def get(self, attribute: str):
+        """Allow this dataclass to be treated like a dictionary.
+
+        This is a work around until the UI fully supports dataclasses
+        at which time this can be removed.
+
+        Default is intentionally not implemented as a safeguard to enure
+        this is not more work than necessary to remove in the future
+        and will only return attributes in existence.
+
+        :param attribute: The attribute to get
+        :returns: The gotten attribute
+        """
+        return getattr(self, attribute)
+
+    def __lt__(self, other):
+        """Compare based on name, called by sort, sorted.
+
+        :param other: The entry to compare this to
+        :returns: Indication of less than the other
+        """
+        return self.name < other.name
+
+    @classmethod
+    def for_settings_file(cls: Type[TEnt], internals: SimpleNamespace) -> TEnt:
+        """Create an ``_HRSettingsEntry`` containing the details for the settings file.
+
+        :param internals: The internal storage for settings information
+        :returns: The settings file entry
+        """
+        description = (
+            "The path to the current settings file. Possible locations are"
+            " {CWD}/ansible-navigator.{ext} or {HOME}/.ansible-navigator.{ext}"
+            " where ext is yml, yaml or json."
+        )
+        return cls(
+            choices=[],
+            current_settings_file=internals.settings_file_path or C.NONE.value,
+            current_value=internals.settings_file_path or C.NONE.value,
+            default=internals.settings_source is C.NONE,
+            default_value=C.NONE.value,
+            description=description,
+            name="Current settings file",
+            env_var="ANSIBLE_NAVIGATOR_CONFIG",
+            settings_file_sample="Not applicable",
+            source=internals.settings_source.value,
+        )
+
+    @classmethod
+    def from_settings_entry(
+        cls: Type[TEnt],
+        application_name_dashed: str,
+        entry: SettingsEntry,
+        settings_file_path: str,
+    ) -> TEnt:
+        """Create an ``_HRSettingsEntry`` containing the details for one settings entry.
+
+        :param application_name_dashed: The application name, dashed
+        :param entry: A settings entry
+        :param settings_file_path: The path to the settings file
+        :return: The settings file entry
+        """
+        cli_parameters = PresentableCliParameters.from_cli_params(
+            cli_parameters=entry.cli_parameters,
+            name_dashed=entry.name_dashed,
+        )
+
+        env_var = entry.environment_variable(application_name_dashed)
+        entry_value_resolved = entry.value.resolved
+
+        path = entry.settings_file_path(application_name_dashed)
+        settings_file_sample = create_settings_file_sample(path, placeholder="<------")
+
+        result = cls(
```

Comment:
Please name the variable referring to things they contain/represent. This `result` name isn't any more helpful than `_`, or `xxx`, or `i`.

Reply:
OTOH, I'd argue that having such throw-away variables is rather pointless and is discouraged by proper linters.